### PR TITLE
Create 2130-maximum-twin-sum-of-a-linked-list.js

### PR DESCRIPTION
**Pull request by aadil42**
Added solution for maximum-twin-sum-of-a-linked-list in JS.

File(s) Added: 2130-maximum-twin-sum-of-a-linked-list.js
Language(s) Used: JavaScript
Submission URL: https://leetcode.com/problems/maximum-twin-sum-of-a-linked-list/submissions/992780842/

### CODE REVIEW

File: javascript/2130-maximum-twin-sum-of-a-linked-list.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    const next = head.next;
+    head.next = temp;
+    temp = head;
+    head = next;
+    count++;
+  }
+  return temp;
+};
+
+var pairSum = function (head) {
```
Comment (aakhtar3) [inline javascript/2130-maximum-twin-sum-of-a-linked-list.js:37]: Suggestion: move your main function to the top of file with complexity comments. And abstract your main function

```js
const length = getLength();
const leftPointer = getLeft();
const rightPointer = getRight();

return getMax();
```

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/2130-maximum-twin-sum-of-a-linked-list.js (hunk 1 above)
Comment (aadil42) [reply javascript/2130-maximum-twin-sum-of-a-linked-list.js:37]: I have moved the main function to the top of the file with time complexity comments. What do you mean by "abstract your main function"?

Comment (aakhtar3) [reply javascript/2130-maximum-twin-sum-of-a-linked-list.js:15]: The suggestion I posted is an example. The purpose is to back down the algorithm. into modular algorithms. Linked List has common patterns to get the mid, reverse, find cycle, etc. One or more of these patterns are always used in solving problems and abstracting your main function to invoke the patterns.

Comment (aadil42) [reply javascript/2130-maximum-twin-sum-of-a-linked-list.js:15]: Haven't I modularized the code by writing _llLength_ and _reverseLL_ functions? What else would have me modularized? Any suggestion would be helpful.

Comment (aakhtar3) [reply javascript/2130-maximum-twin-sum-of-a-linked-list.js:15]: You can extract out 

const rightPointer = getRightPointer()
const leftPointer = reverseLL(head, mid);
const max = getMax();

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/2130-maximum-twin-sum-of-a-linked-list.js (hunk 1 above)
Comment (aadil42) [reply javascript/2130-maximum-twin-sum-of-a-linked-list.js]: @aakhtar3, I did modularized it as suggested. Please check. Here's submission [link](https://leetcode.com/problems/maximum-twin-sum-of-a-linked-list/submissions/1042826145/)

**Review by aakhtar3 — APPROVED** [review]

**Pull request merged**